Added nugget to ingot and ingot to block recipes for EIO materials

Flamebom opened the pull request:
# Description
Added shapeless nugget to ingot and shapeless ingot to block recipes for materials with datagen.
# Checklist:

- [x] My code follows the style guidelines of this project (.editorconfig, most IDEs will use this for you)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

Flamebom committed:
Added missing copper alloy items

Flamebom committed:
Added more consistent shading

Flamebom committed:
Merge remote-tracking branch 'upstream/dev/1.19.x' into dev/1.19.x

Flamebom committed:
nuggets to ingots and ingots to blocks

justliliandev commented on src/main/java/com/enderio/base/data/recipe/MaterialRecipes.java:
```diff
@@ -269,13 +287,24 @@ private void blockToIngots(Consumer<FinishedRecipe> recipeConsumer, Item ingot,
             .unlockedBy("has_ingredient", InventoryChangeTrigger.TriggerInstance.hasItems(ingot))
             .save(recipeConsumer);
     }
+    private void nuggetstoIngot(Consumer<FinishedRecipe> recipeConsumer, Item ingot, Item nugget) {
```
```suggestion
    private void nuggetsToIngot(Consumer<FinishedRecipe> recipeConsumer, Item ingot, Item nugget) {
```

Flamebom replied:
> it might also be useful to merge all 4 recipes block<->ingot<->nugget into one helper accepting nugget, ingot, block item

Added

justliliandev reviewed (changes requested):
it might also be useful to merge all 4 recipes block<->ingot<->nugget into one helper accepting nugget, ingot, block item

Flamebom committed:
Added method for material conversion

justliliandev commented on src/main/java/com/enderio/base/data/recipe/MaterialRecipes.java:
```diff
@@ -281,31 +254,24 @@ protected void buildCraftingRecipes(Consumer<FinishedRecipe> recipeConsumer) {
-    
+	private void makeMaterialRecipes(Consumer<FinishedRecipe> recipeConsumer, Item ingot, Item nugget, Block block ) {
+		ShapelessRecipeBuilder.shapeless(ingot, 9)
+		.requires(block.asItem())
```
Add Tab on all those calls on the builder

Flamebom replied:
also done

justliliandev replied:
The shaped calls aren't tabbed correctly

Flamebom committed:
Added tabs and changed recipes

Added tabs to the recipe builder
changed recipes from shapeless to shaped to merge them together

Flamebom committed:
fixed indenting (hopefully?)

Rover656 reviewed (approved):
Looks good to me, thanks :)
